pyglider/utils.py

```diff
@@ -361,6 +362,21 @@ def nmea2deg(nmea):
            np.sign(nmea) * np.remainder(np.abs(nmea), 100) / 60)
     return deg
 
+
+def oxygen_concentration_correction(data, ncvar):
+    oxy_yaml = ncvar['oxygen_concentration']
+    if 'reference_salinity' not in oxy_yaml.keys():
+        _log.warning('No reference_salinity found in oxygen deployment yaml. Assuming reference salinity of 0 psu')
+        ref_sal = 0
+    else:
+        ref_sal = float(oxy_yaml['reference_salinity'])
+    sa = gsw.SA_from_SP(data['salinity'], data['pressure'], data['longitude'], data['latitude'])
```

Comment:
OK, so maybe I should have read the gsw PR more carefully.  What are longitude and latitude doing in this call?

Reply:
Ah good catch Jody. While the glider data should only store SP, of course the `gsw_O2sol()` function requires SA and CT (which in turn require lon/lat). See:

https://www.teos-10.org/pubs/gsw/html/gsw_O2sol.html

GSW is great but man sometimes it fixes something that wasn't really broken in the first place ...

Reply:
Oh, I see, so `data['salinity']` is indeed practical salinity, or at least some version that is not geographically corrected, but O2sol wants the correction.  Though I'm not sure we want to give it the corrected version?  What happens if you just use SP here?  I suspect the difference is less than the accuracy of most optodes.    

I've not been convinced GSW is great.  If your values depend on a spatial correction, one hopes they are very small...  

I think that means there needs to be info added to the attributes as to how this is processed.

Reply:
I've had a look at the GSW code, it is indeed doing a geographical correction to SA, which is then used in calculation oxygen concentration. I've tested changing the lon and lat to various oceans (Atlantic, Pacific, Ross Sea, Grreenland Sea & Baltic). The changes to oxygen are way below the noise level of the optodes. Only the Baltic had changes I could see when zoomed right in. The others are near identical.

![comp_zoom](https://user-images.githubusercontent.com/28703282/145690658-f182bcc4-d484-4f1e-8d1d-f4e17307ca69.png)

Here's the mean oxygen concetration values. This is using the example-seaeplorer data

uncorr mean: 180.39353891647335
true mean: 145.7232575360533
trop_atl mean: 145.72527999183677
trop_sth_pac mean: 145.72592724849966
ross_sea mean: 145.72227471307164
baltic mean: 145.62095778937243

Reply:
Here's a gist of the code I used in the above calculations

https://gist.github.com/callumrollo/d17cfef02b5830c23f2c21ebe46ecdd8

Reply:
I'll update the attributes to reflect the processing

Reply:
OK, but my tendency would be to use the simplest processing possible, which means to use SP, not SA - if it really makes no practical difference, its easier to back out.